package/assets/middleware.ts

```diff
@@ -11,7 +11,7 @@ const extractLocaleFromUrl = (pathname: string) => {
 		} else if (options.strategy === "prefixExceptDefault") {
 			if (
 				locale !== options.defaultLocale &&
-				pathname.startsWith(`/${locale}/`)
+				pathname.startsWith(`/${locale}`)
```

Comment:
@amosbastian perhaps you can try to use 

```js
const locale = 'fr';
const regex = new RegExp(`^/${locale}(/|$|/[^/]+)`);

console.log(regex.test('/fr'));      // true
console.log(regex.test('/fr/'));     // true
console.log(regex.test('/fr/boo'));  // true
console.log(regex.test('/freedom')); // false
```

instead of `pathname.startsWith`